src/annotation/index.ts

```diff
@@ -814,6 +826,33 @@ export const annotationTypeHandlers: Record<
       callback(annotation.pointA, false);
       callback(annotation.pointB, false);
     },
+    defaultProperties(annotation: Line, scales: Float64Array) {
+      const properties: AnnotationNumericPropertySpec[] = [];
+      const values: number[] = [];
+      const rank = scales.length;
+      if (
+        rank === annotation.pointA.length &&
+        rank === annotation.pointB.length
+      ) {
+        properties.push({
+          type: "float32",
+          identifier: "Length",
+          default: 0,
+          description: "Length of the line annotation in nanometers",
+          format: formatLength,
+        });
+        let length = 0;
+        for (let dim = 0; dim < rank; dim++) {
+          length +=
+            (((annotation.pointA[dim] - annotation.pointB[dim]) / 1e-9) *
```

Comment:
This shouldn't always be nanometers --- it should take into account the units set in the coordinate space.

If all dimensions have the same base unit, e.g. meter, then you could just choose an appropriate scale automatically as in the scale bar.

If there are mixed units, e.g. meters and seconds, then probably just don't display the length at all.

Reply:
I updated it to use globalCoordinateSpace.units and using `ALLOWED_UNITS` from scale_bar.ts to disable the property. If any unit is a non-length unit, it won't add the property. Also cleaned up the code